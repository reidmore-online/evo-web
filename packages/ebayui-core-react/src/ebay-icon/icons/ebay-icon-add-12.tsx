/* This file is autogenerated, do not edit - it will be overwritten anyway */

import React from "react";
import EbayIcon from "../icon";
import type { EbayIconComponent } from "./types";

const SYMBOL = `<symbol viewBox="0 0 12 12" id="icon-add-12"><path fill-rule="evenodd" clip-rule="evenodd" d="M6.75 1.75a.75.75 0 0 0-1.5 0v3.5h-3.5a.75.75 0 0 0 0 1.5h3.5v3.5a.75.75 0 0 0 1.5 0v-3.5h3.5a.75.75 0 0 0 0-1.5h-3.5v-3.5Z"/></symbol>`;

export const EbayIconAdd12: EbayIconComponent = props => (
    <EbayIcon {...props} name="add12" __type="icon" __symbol={SYMBOL} />
);