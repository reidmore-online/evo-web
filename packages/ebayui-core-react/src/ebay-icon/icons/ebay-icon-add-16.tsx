/* This file is autogenerated, do not edit - it will be overwritten anyway */

import React from "react";
import EbayIcon from "../icon";
import type { EbayIconComponent } from "./types";

const SYMBOL = `<symbol viewBox="0 0 16 16" id="icon-add-16"><path d="M14 7H9V2a1 1 0 0 0-2 0v5H2a1 1 0 0 0 0 2h5v5a1 1 0 1 0 2 0V9h5a1 1 0 1 0 0-2Z"/></symbol>`;

export const EbayIconAdd16: EbayIconComponent = props => (
    <EbayIcon {...props} name="add16" __type="icon" __symbol={SYMBOL} />
);