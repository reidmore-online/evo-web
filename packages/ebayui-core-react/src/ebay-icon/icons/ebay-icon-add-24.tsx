/* This file is autogenerated, do not edit - it will be overwritten anyway */

import React from "react";
import EbayIcon from "../icon";
import type { EbayIconComponent } from "./types";

const SYMBOL = `<symbol viewBox="0 0 24 24" id="icon-add-24"><path d="M21 11h-8V3a1 1 0 0 0-2 0v8H3a1 1 0 0 0 0 2h8v8a1 1 0 0 0 2 0v-8h8a1 1 0 0 0 0-2Z"/></symbol>`;

export const EbayIconAdd24: EbayIconComponent = props => (
    <EbayIcon {...props} name="add24" __type="icon" __symbol={SYMBOL} />
);