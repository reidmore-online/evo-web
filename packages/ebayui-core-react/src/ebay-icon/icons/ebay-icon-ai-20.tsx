/* This file is autogenerated, do not edit - it will be overwritten anyway */

import React from "react";
import EbayIcon from "../icon";
import type { EbayIconComponent } from "./types";

const SYMBOL = `<symbol viewBox="0 0 20 20" id="icon-ai-20"><path fill-rule="evenodd" clip-rule="evenodd" d="M10 6.303A6.32 6.32 0 0 1 6.303 10 6.32 6.32 0 0 1 10 13.697 6.32 6.32 0 0 1 13.697 10 6.32 6.32 0 0 1 10 6.303Zm1.013-4.484c-.234-1.092-1.792-1.092-2.026 0L8.306 4.99A4.317 4.317 0 0 1 4.99 8.306l-3.172.681c-1.092.234-1.092 1.792 0 2.026l3.172.681a4.317 4.317 0 0 1 3.315 3.315l.681 3.172c.234 1.092 1.792 1.092 2.026 0l.681-3.172a4.317 4.317 0 0 1 3.315-3.315l3.172-.681c1.092-.234 1.092-1.792 0-2.026l-3.172-.681a4.317 4.317 0 0 1-3.315-3.315l-.681-3.172Z"/></symbol>`;

export const EbayIconAi20: EbayIconComponent = props => (
    <EbayIcon {...props} name="ai20" __type="icon" __symbol={SYMBOL} />
);