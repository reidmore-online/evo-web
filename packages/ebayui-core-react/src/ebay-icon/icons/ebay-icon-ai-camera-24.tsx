/* This file is autogenerated, do not edit - it will be overwritten anyway */

import React from "react";
import EbayIcon from "../icon";
import type { EbayIconComponent } from "./types";

const SYMBOL = `<symbol viewBox="0 0 24 24" id="icon-ai-camera-24"><path d="M12.75 2a1 1 0 1 1 0 2H9.441a1 1 0 0 0-.948.684L7.951 6.31a1.01 1.01 0 0 1-.958.69H3a1 1 0 0 0-1 1v11l.005.102A1 1 0 0 0 3 20h18a1 1 0 0 0 1-1v-7.5a1 1 0 1 1 2 0V19a3 3 0 0 1-3 3H3a3 3 0 0 1-2.996-2.846L0 19V8a3 3 0 0 1 3-3h3.28l.316-.948A3 3 0 0 1 9.44 2h3.309ZM12 8.25a4.75 4.75 0 1 1 0 9.5 4.75 4.75 0 0 1 0-9.5Zm0 2a2.75 2.75 0 1 0 0 5.5 2.75 2.75 0 0 0 0-5.5Zm6.125-8.543c.202-.943 1.548-.943 1.75 0l.352 1.636a1.864 1.864 0 0 0 1.43 1.43l1.636.352c.943.202.943 1.548 0 1.75l-1.636.352a1.864 1.864 0 0 0-1.43 1.431l-.352 1.635c-.203.943-1.547.943-1.75 0l-.352-1.635a1.865 1.865 0 0 0-1.431-1.431l-1.635-.352c-.943-.202-.943-1.548 0-1.75l1.635-.352a1.864 1.864 0 0 0 1.431-1.43l.352-1.636Z"/></symbol>`;

export const EbayIconAiCamera24: EbayIconComponent = props => (
    <EbayIcon {...props} name="aiCamera24" __type="icon" __symbol={SYMBOL} />
);