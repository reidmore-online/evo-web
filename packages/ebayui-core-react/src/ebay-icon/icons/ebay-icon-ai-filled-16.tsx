/* This file is autogenerated, do not edit - it will be overwritten anyway */

import React from "react";
import EbayIcon from "../icon";
import type { EbayIconComponent } from "./types";

const SYMBOL = `<symbol viewBox="0 0 16 16" id="icon-ai-filled-16"><path d="M7.212 1.637c.182-.85 1.394-.85 1.576 0l.53 2.467a3.358 3.358 0 0 0 2.578 2.578l2.467.53c.85.182.85 1.394 0 1.576l-2.467.53a3.358 3.358 0 0 0-2.578 2.578l-.53 2.467c-.182.85-1.394.85-1.576 0l-.53-2.467a3.358 3.358 0 0 0-2.578-2.578l-2.467-.53c-.85-.182-.85-1.394 0-1.576l2.467-.53a3.358 3.358 0 0 0 2.578-2.578l.53-2.467Z"/></symbol>`;

export const EbayIconAiFilled16: EbayIconComponent = props => (
    <EbayIcon {...props} name="aiFilled16" __type="icon" __symbol={SYMBOL} />
);