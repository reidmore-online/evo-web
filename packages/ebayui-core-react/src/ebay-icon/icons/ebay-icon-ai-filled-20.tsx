/* This file is autogenerated, do not edit - it will be overwritten anyway */

import React from "react";
import EbayIcon from "../icon";
import type { EbayIconComponent } from "./types";

const SYMBOL = `<symbol viewBox="0 0 20 20" id="icon-ai-filled-20"><path d="M8.987 1.819c.234-1.092 1.792-1.092 2.026 0l.681 3.172a4.317 4.317 0 0 0 3.315 3.315l3.172.681c1.092.234 1.092 1.792 0 2.026l-3.172.681a4.317 4.317 0 0 0-3.315 3.315l-.681 3.172c-.234 1.092-1.792 1.092-2.026 0l-.681-3.172a4.317 4.317 0 0 0-3.315-3.315l-3.172-.681c-1.092-.234-1.092-1.792 0-2.026l3.172-.681A4.317 4.317 0 0 0 8.306 4.99l.681-3.172Z"/></symbol>`;

export const EbayIconAiFilled20: EbayIconComponent = props => (
    <EbayIcon {...props} name="aiFilled20" __type="icon" __symbol={SYMBOL} />
);