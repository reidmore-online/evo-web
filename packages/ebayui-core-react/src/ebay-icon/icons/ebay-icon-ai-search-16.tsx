/* This file is autogenerated, do not edit - it will be overwritten anyway */

import React from "react";
import EbayIcon from "../icon";
import type { EbayIconComponent } from "./types";

const SYMBOL = `<symbol viewBox="0 0 16 16" id="icon-ai-search-16"><path d="M5.893 1.033a1 1 0 0 1 .218 1.988 3.5 3.5 0 1 0 3.3 5.424l.06-.082a1 1 0 0 1 1.6 1.196l-.042.056 3.678 3.678.068.076a1 1 0 0 1-1.406 1.406l-.076-.068-3.678-3.678A5.463 5.463 0 0 1 6.5 12a5.5 5.5 0 0 1-.607-10.967Zm4.54-.325c.132-.61 1.002-.61 1.133 0l.229 1.063c.1.468.466.834.934.934l1.063.229c.61.13.61 1.001 0 1.132l-1.063.229c-.468.1-.834.466-.934.934l-.229 1.063c-.13.61-1.001.61-1.132 0l-.229-1.063a1.217 1.217 0 0 0-.934-.934l-1.063-.229c-.61-.13-.61-1.001 0-1.132l1.063-.229c.468-.1.834-.466.934-.934l.229-1.063Z"/></symbol>`;

export const EbayIconAiSearch16: EbayIconComponent = props => (
    <EbayIcon {...props} name="aiSearch16" __type="icon" __symbol={SYMBOL} />
);