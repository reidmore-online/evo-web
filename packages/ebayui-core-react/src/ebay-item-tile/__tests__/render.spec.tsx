import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import {
    EbayItemTile,
    EbayItemTileSupertitle,
    EbayItemTileTitle,
    EbayItemTileSubtitle,
    EbayItemTileDescription,
    EbayItemTileAction,
} from "../";
import { EbayIconHeart16 } from "../../ebay-icon/icons/ebay-icon-heart-16";

describe("<EbayItemTile>", () => {
    it("renders component with action", () => {
        const { asFragment } = render(
            <EbayItemTile
                file={{
                    name: "file-name.jpg",
                    type: "image",
                    src: "https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg",
                }}
            >
                <EbayItemTileAction aria-label="action-label" icon={<EbayIconHeart16 />} />
                <EbayItemTileSupertitle>Time Sensitive</EbayItemTileSupertitle>
                <EbayItemTileTitle href="/collection">Apple iPhone 11 Pro Max </EbayItemTileTitle>
                <EbayItemTileSubtitle>256GB Space Gray</EbayItemTileSubtitle>
                <EbayItemTileDescription className="price">$29.99</EbayItemTileDescription>
                <EbayItemTileDescription as="div">
                    <a href="https://ebay.com">Buy it now</a>
                </EbayItemTileDescription>
                <EbayItemTileDescription>Free shipping</EbayItemTileDescription>
            </EbayItemTile>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
    it("renders component without action", () => {
        const { asFragment } = render(
            <EbayItemTile
                file={{
                    name: "file-name.jpg",
                    type: "image",
                    src: "https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg",
                }}
            >
                <EbayItemTileSupertitle>Time Sensitive</EbayItemTileSupertitle>
                <EbayItemTileTitle href="/collection">Apple iPhone 11 Pro Max </EbayItemTileTitle>
                <EbayItemTileSubtitle>256GB Space Gray</EbayItemTileSubtitle>
                <EbayItemTileDescription className="price">$29.99</EbayItemTileDescription>
                <EbayItemTileDescription as="div">
                    <a href="https://ebay.com">Buy it now</a>
                </EbayItemTileDescription>
                <EbayItemTileDescription>Free shipping</EbayItemTileDescription>
            </EbayItemTile>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
    it("renders component without secondary section", () => {
        const { asFragment } = render(
            <EbayItemTile
                file={{
                    name: "file-name.jpg",
                    type: "image",
                    src: "https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg",
                }}
            >
                <EbayItemTileSupertitle>Time Sensitive</EbayItemTileSupertitle>
                <EbayItemTileDescription className="price">$29.99</EbayItemTileDescription>
                <EbayItemTileDescription as="div">
                    <a href="https://ebay.com">Buy it now</a>
                </EbayItemTileDescription>
                <EbayItemTileDescription>Free shipping</EbayItemTileDescription>
            </EbayItemTile>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
    it("renders component without description", () => {
        const { asFragment } = render(
            <EbayItemTile
                file={{
                    name: "file-name.jpg",
                    type: "image",
                    src: "https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg",
                }}
            >
                <EbayItemTileSupertitle>Time Sensitive</EbayItemTileSupertitle>
                <EbayItemTileTitle href="/collection">Apple iPhone 11 Pro Max </EbayItemTileTitle>
                <EbayItemTileSubtitle>256GB Space Gray</EbayItemTileSubtitle>
            </EbayItemTile>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
