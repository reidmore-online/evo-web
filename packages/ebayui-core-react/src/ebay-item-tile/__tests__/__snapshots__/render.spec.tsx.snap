// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<EbayItemTile> renders component with action 1`] = `
<DocumentFragment>
  <div
    class="item-tile"
  >
    <div
      class="item-tile__header"
    >
      <div
        class="file-preview-card"
      >
        <div
          class="file-preview-card__body"
        >
          <img
            alt="file-name.jpg"
            class="file-preview-card__asset"
            src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg"
          />
          <button
            aria-label="action-label"
            class="file-preview-card__action icon-btn"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-heart-16"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      class="item-tile__body"
    >
      <div
        class="item-tile__section-primary"
      >
        Time Sensitive
      </div>
      <div
        class="item-tile__section-secondary"
      >
        <a
          class="item-tile__title"
          href="/collection"
        >
          Apple iPhone 11 Pro Max 
        </a>
        <div
          class="item-tile__subtitle"
        >
          256GB Space Gray
        </div>
      </div>
      <div
        class="item-tile__section-tertiary"
      >
        <p
          class="price"
        >
          $29.99
        </p>
        <div>
          <a
            href="https://ebay.com"
          >
            Buy it now
          </a>
        </div>
        <p>
          Free shipping
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<EbayItemTile> renders component without action 1`] = `
<DocumentFragment>
  <div
    class="item-tile"
  >
    <div
      class="item-tile__header"
    >
      <div
        class="file-preview-card"
      >
        <div
          class="file-preview-card__body"
        >
          <img
            alt="file-name.jpg"
            class="file-preview-card__asset"
            src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg"
          />
        </div>
      </div>
    </div>
    <div
      class="item-tile__body"
    >
      <div
        class="item-tile__section-primary"
      >
        Time Sensitive
      </div>
      <div
        class="item-tile__section-secondary"
      >
        <a
          class="item-tile__title"
          href="/collection"
        >
          Apple iPhone 11 Pro Max 
        </a>
        <div
          class="item-tile__subtitle"
        >
          256GB Space Gray
        </div>
      </div>
      <div
        class="item-tile__section-tertiary"
      >
        <p
          class="price"
        >
          $29.99
        </p>
        <div>
          <a
            href="https://ebay.com"
          >
            Buy it now
          </a>
        </div>
        <p>
          Free shipping
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<EbayItemTile> renders component without description 1`] = `
<DocumentFragment>
  <div
    class="item-tile"
  >
    <div
      class="item-tile__header"
    >
      <div
        class="file-preview-card"
      >
        <div
          class="file-preview-card__body"
        >
          <img
            alt="file-name.jpg"
            class="file-preview-card__asset"
            src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg"
          />
        </div>
      </div>
    </div>
    <div
      class="item-tile__body"
    >
      <div
        class="item-tile__section-primary"
      >
        Time Sensitive
      </div>
      <div
        class="item-tile__section-secondary"
      >
        <a
          class="item-tile__title"
          href="/collection"
        >
          Apple iPhone 11 Pro Max 
        </a>
        <div
          class="item-tile__subtitle"
        >
          256GB Space Gray
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<EbayItemTile> renders component without secondary section 1`] = `
<DocumentFragment>
  <div
    class="item-tile"
  >
    <div
      class="item-tile__header"
    >
      <div
        class="file-preview-card"
      >
        <div
          class="file-preview-card__body"
        >
          <img
            alt="file-name.jpg"
            class="file-preview-card__asset"
            src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg"
          />
        </div>
      </div>
    </div>
    <div
      class="item-tile__body"
    >
      <div
        class="item-tile__section-primary"
      >
        Time Sensitive
      </div>
      <div
        class="item-tile__section-tertiary"
      >
        <p
          class="price"
        >
          $29.99
        </p>
        <div>
          <a
            href="https://ebay.com"
          >
            Buy it now
          </a>
        </div>
        <p>
          Free shipping
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
